client/js/loading-slow-alert.js

```diff
@@ -8,40 +8,53 @@
  *    so that the timeout can be triggered while slow JS is loading
  */
 
-setTimeout(function() {
-	var element = document.getElementById("loading-slow");
+function displayReload() {
+	var loadingReload = document.getElementById("loading-reload");
+	if (loadingReload) {
+		loadingReload.style.display = "block";
+	}
+}
+
+var loadingSlowTimeout = setTimeout(function() {
+	var loadingSlow = document.getElementById("loading-slow");
 
-	if (element) {
-		element.style.display = "block";
+	if (loadingSlow) {
+		loadingSlow.style.display = "block";
 	}
+
+	displayReload();
 }, 5000);
 
-document.getElementById("loading-slow-reload").addEventListener("click", function() {
+document.getElementById("loading-reload").addEventListener("click", function() {
 	location.reload();
 });
 
-window.g_LoungeErrorHandler = function LoungeErrorHandler(error) {
+window.g_LoungeErrorHandler = function LoungeErrorHandler(e) {
 	var title = document.getElementById("loading-title");
 	title.textContent = "An error has occured";
 
-	title = document.getElementById("loading-page-message");
-	title.textContent = "An error has occured that prevented the client from loading correctly.";
+	var message = document.getElementById("loading-page-message");
+	message.textContent = "An error has occured that prevented the client from loading correctly.";
 
 	var summary = document.createElement("summary");
 	summary.textContent = "More details";
 
-	if (error instanceof ErrorEvent) {
-		error = error.message + "\n\n" + error.stack + "\n\nView developer tools console for more information and a better stacktrace.";
+	var data = document.createElement("pre");
+	if (e instanceof ErrorEvent) {
+		data.textContent = e.error && e.error.stack ? e.error.stack : e.message;
```

Comment:
I didn't include stacktrace because it's completely useless without sourcemap loaded in, and that only loads in with devtools open before the error happens, so it's effectively very useless and misleading.

Reply:
Interesting. You did include `error.stack` and that was always `undefined` (at least for me), so I figured you wanted the stack trace. Want me to remove it?

Reply:
> that only loads in with devtools open before the error happens

Odd, because I'm trying everything to cause this, even loading in private browsing and such, and it always shows up. Also I have a fallback if the stack is empty.

But I get your overall point. Want me to remove it?

Reply:
I don't understand how it could possibly work since sourcemaps are loaded by devtools. Looking at other error reporters, they either post-process errors on server, or load sourcemap as specified in JS themselves.

So yeah, it's better off removing it since we're not making a fully fledged error reporter here.

Reply:
> I don't understand how it could possibly work since sourcemaps are loaded by devtools.

I don't know, mate, it just works :man_shrugging: 
But you'll notice that on the screenshot it shows the stacktrace of the built JS bundle, not the original source. That makes it less interesting for sure.

Will remove.

Reply:
Done:

<img width="443" alt="screen shot 2017-12-17 at 23 53 36" src="https://user-images.githubusercontent.com/113730/34090599-38695c86-e386-11e7-95e6-4a141ba76bcb.png">

Very glad you added this page overall. I had my concerns at first because of misunderstanding, but it makes the experience more pro and less frustrating 👏